src/braket/experimental/autoqasm/api.py

```diff
@@ -227,25 +218,33 @@ def _convert_program(
         user_config (UserConfig): User-specified settings that influence program building
         args (List[Any]): Arguments passed to the program when called.
         kwargs (Dict[str, Any]): Keyword arguments passed to the program when called.
+        is_subroutine (bool): If the function corresponds to a subroutine or main.
+
 
     Returns:
         Union[Program, Optional[Var]]: The converted program, if this is a top-level call
         to the conversion process. Or, the oqpy variable returned from the converted function,
         if this is a subroutine conversion.
     """
-    # We will convert this function as a subroutine if we are already inside an
-    # existing conversion process (i.e., this is a subroutine call).
-    convert_as_subroutine = aq_program.in_active_program_conversion_context()
+    if is_subroutine and not aq_program.in_active_program_conversion_context():
+        raise errors.AutoQasmTypeError(
+            "Subroutines shouldn't be called directly. Please define an entry point "
+            "function, decorate it with '@autoqasm.main', and call your subroutine "
```

Comment:
I think our docs are consistently using the `aq.` prefix
```suggestion
            "function, decorate it with '@aq.main', and call your subroutine "
```